beets/dbcore/db.py

```diff
@@ -585,38 +585,34 @@ def store(self, fields: Optional[Iterable[str]] = None):
         will be.
         """
         if fields is None:
-            fields = self._fields
+            fields = self._fields.keys()
+        fields = set(fields) - {"id"}
         db = self._check_db()
 
         # Build assignments for query.
-        assignments = []
-        subvars = []
-        for key in fields:
-            if key != "id" and key in self._dirty:
-                self._dirty.remove(key)
-                assignments.append(key + "=?")
-                value = self._type(key).to_sql(self[key])
-                subvars.append(value)
+        dirty_fields = list(fields & self._dirty)
+        self._dirty -= fields
+        assignments = ",".join(f"{k}=?" for k in dirty_fields)
+        subvars = [self._type(k).to_sql(self[k]) for k in dirty_fields]
 
         with db.transaction() as tx:
             # Main table update.
             if assignments:
-                query = "UPDATE {} SET {} WHERE id=?".format(
-                    self._table, ",".join(assignments)
-                )
+                query = f"UPDATE {self._table} SET {assignments} WHERE id=?"
                 subvars.append(self.id)
                 tx.mutate(query, subvars)
 
             # Modified/added flexible attributes.
-            for key, value in self._values_flex.items():
-                if key in self._dirty:
-                    self._dirty.remove(key)
-                    tx.mutate(
-                        "INSERT INTO {} "
-                        "(entity_id, key, value) "
-                        "VALUES (?, ?, ?);".format(self._flex_table),
-                        (self.id, key, value),
-                    )
+            flex_fields = set(self._values_flex.keys())
+            dirty_flex_fields = list(flex_fields & self._dirty)
+            self._dirty -= flex_fields
+            for key in dirty_flex_fields:
+                tx.mutate(
+                    f"INSERT INTO {self._flex_table} "
+                    "(entity_id, key, value) "
+                    "VALUES (?, ?, ?);",
+                    (self.id, key, self._values_flex[key]),
+                )
```

Comment:
This does result in the same values, but not the same ordering because of the conversion to sets. If that will cause problems should be double and triple-checks since this is ultimately being made into SQL commands.

Reply:
Oh, I forgot about the order changing!  I don't _think_ it'll cause issues, but I can actually think of a nicer layout for this implementation in terms of `not in` that does preserve the order.  I'll implement it in its own commit, we'll see what's preferable.

Reply:
Okay, I've rewritten the code now.  I'm confident the SQL assignments do not care about the order fields are set in, so I've stuck with sets.  I traced where the `fields` parameter can come from anyways -- it looks like they are user-set (`beet update -f FIELDS`), in which case we really shouldn't be relying on the order anyways.

Reply:
Wonderful, I'll leave this issue open until I double check myself.